TSOA + YUP

AsianHam opened the pull request:
created routes using tsoa

added data validation via yup

AsianHam committed:
basic tsoa implementation

restructured code to implement the tsoa library

created basic get function for /recipe/:id that returns a static object

moved interface into separate file

created controller and service file

created default tsoa routes file

changed name of old app.ts file to old-app.ts; this file should not be used anymore, kept only for reference

AsianHam committed:
nodemon/express touchup

fixed nodemon not updating on file change

updated express server to include extended option to remove deprecation warning

AsianHam committed:
Yup Implementation

setup yup to validate request body

added invalid id check for put

added proper success responses

AsianHam committed:
Clean up

removed unneeded dependency

removed original app.ts

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+  | "steps"
+>;
+
+dotenv.config();
```
I think you want to run this in app.ts, thats generally where people put it

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+  link: [],
+  metadata: {},
+  ingredients: [],
+  steps: {},
```
Shouldnt this be an array now?

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+
+export async function getRecipe(id: number) {
+  try {
+    const data = await recipeModel.find({ id: id });
```
You can make this one line, no need to create a variable

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+
+export async function getAllRecipe() {
+  try {
+    const data = await recipeModel.find();
```
Same here, no need for a variable

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+
+export async function searchRecipe(ingredient: string) {
+  try {
+    const data = await recipeModel.find({
```
Same here

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,99 @@
+export async function createRecipe(recipeBody: Recipe) {
+  try {
+    const data = new recipeModel(recipeBody);
+    const dataToSave = await data.save();
```
Same here

AsianHam committed:
update git ignore

removed swagger.json and routes.ts

AsianHam committed:
code clean up

moved dotenv configuration to app.ts

removed unnecessary variables

AsianHam committed:
schema update

updated steps to represent an array, not an object

Jacksondr5 commented on package.json:
```diff
@@ -4,27 +4,30 @@
     "tslint": "^6.1.3",
-    "typescript": "^4.6.4"
+    "typescript": "^4.7.3",
+    "yup": "^0.32.11"
```
Move this to the dependencies object

Jacksondr5 commented on package.json:
```diff
@@ -4,27 +4,30 @@
+    "@types/multer": "^1.4.7",
+    "@types/node": "^17.0.40",
+    "concurrently": "^7.2.1",
+    "cross-env": "^7.0.3",
```
Move this to dependencies

Jacksondr5 commented on src/recipeService.ts:
```diff
@@ -0,0 +1,94 @@
+
+export async function getRecipe(id: number) {
+  try {
+    return await recipeModel.find({ id: id });
```
This should be findOne, because find retruns an array and thats not what you want here

AsianHam committed:
pr fixes

moved some devdependencies to depencies

changed a function to find a singular result
